### PR TITLE
chore: upgrade `bazel` → `7.0.0`

### PR DESCRIPTION
**Pull request by sgammon**
## New Image Pull Request Template

Not a new image, but I will fill this form out anyway as it looks to be a requirement

### Image Size

- [x] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes:
It's just Bazel 7. The image is smaller because I removed the JVM which was pre-installed. No local JVM is needed to run Bazel, as Bazel itself contains an embedded JVM which is uses to bootstrap execution. Toolchain and runtime JVMs can be added by the user and consumed hermetically (and should be).

### Image Vulnerabilities

- [ ] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).
- [x] No idea, I'm updating an existing image which needs a new release

Notes:

### Image Tagging

- [ ] The image is _not_ tagged with version tags.
- [x] The image is tagged with :latest
- [ ] The image is not tagged with :latest (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [ ] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).
- [x] I don't know what a kind cluster is :(

Notes:

### Basic Testing - Package/Application

- [ ] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:
I'm not sure how to test images built with Terraform. I'll try to figure out how to do this.

### Helm

- [ ] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [x] A Helm chart was not provided.

Notes:

### Processor Architectures

- [ ] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [ ] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [ ] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:
Bazel is just a tool so it doesn't need a Helm chart.

### Environment Testing + Documentation

- [ ] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:
When I figure out how to build the image I'm happy to test it.

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:
Bazel `7.0.0` is the latest LTS release version.

### Dev Tag Availability

- [ ] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [x] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:
There's not, but it would be great if there was one. I'm happy to add one if I can get some direction here.

### Access Control + Authentication

- [ ] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).
- [x] Unmodified (existing image)

### ENTRYPOINT

- [ ] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).
- [x] Unmodified (existing image)

### CMD

- [ ] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]
- [x] Unmodified (existing image)

### Environment Variables

- [ ] Environment variables added.
- [ ] Environment variables not added and not required.
- [x] Unmodified (existing image)

### SIGTERM

- [ ] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)
- [x] Unmodified (existing image)

### Logs

- [ ] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.
- [x] Unmodified (existing image)

### Documentation - README

- [ ] A README file has been provided and it follows the README template.
- [x] Unmodified (existing image)

### CODE REVIEW

**Comment by sgammon** [comment]
I authored this commit through Github's web interface, hence the DCO error. I'll pull it down and sign it with GPG and for DCO.

File: images/bazel/main.tf [hunk 1]
```diff
@@ -10,7 +10,7 @@ variable "target_repository" {
 
 module "config" {
   source         = "./config"
-  extra_packages = ["bazel-6", "openjdk-17", "openjdk-17-default-jvm"]
+  extra_packages = ["bazel-7"]
```
Comment (imjasonh) [inline images/bazel/main.tf:13]:
```suggestion
  extra_packages = ["bazel-7", "openjdk-17", "openjdk-17-default-jvm"]
```

**Comment by imjasonh** [comment]
Thanks for this! The image still needs openjdk-17, and for it to be the default JVM in the image, so this should be a simple change.

**Comment by imjasonh** [comment]
Or, should it include openjdk-21 instead?

**Comment by sgammon** [comment]
@imjasonh I'm unsure why the JDK is included? Bazel embeds its own JDK. Is there another reason it should be there? Thank you for the quick review.

**Comment by sgammon** [comment]
We're currently nearly done testing this in our workflow so I can report back how it performs.

**Comment by sgammon** [comment]
@imjasonh Wow, hm, you're absolutely right. I guess it still does require a JDK to be installed. I was under the impression it did not, based on a comment from the Bazel team.

In any case, I added `openjdk-21` as the default and signed the commit.

**Comment by sgammon** [comment]
We're now using this image effectively in our workflows :)

**Review by imjasonh — APPROVED** [review]

**Comment by sgammon** [comment]
@imjasonh Failure says:

> Error: Test failed. Warning: Applied changes may be incomplete

Any hints? I'm rather new to Terraform used in this way. I'm happy to push changes to get the tests passing.

**Comment by imjasonh** [comment]
> @imjasonh Failure says:
> 
> > Error: Test failed. Warning: Applied changes may be incomplete
> 
> Any hints? I'm rather new to Terraform used in this way. I'm happy to push changes to get the tests passing.

The real error is actually unfortunately hidden in a collapsed section of the next step:

```
  ERROR: Error computing the main repository mapping: MODULE.bazel cannot be created at /work/MODULE.bazel: /work/MODULE.bazel (Permission denied)
```

https://github.com/chainguard-images/images/actions/runs/7498480873/job/20667897274?pr=2062#step:8:78

Does bazel 7 require being run as root?

**Comment by sgammon** [comment]
@imjasonh Oh, I see. This is so annoying. Since the transition to `bzlmod`, Bazel will auto-create a `MODULE.bazel` file, unless instructed not to via `--noenable_bzlmod`. It creates it and fills it with a comment to get the user started in their transition away from `WORKSPACE` (blah blah).

Point is, it's trying to write to disk for a user tutorial step, when I call `bazel version` (which I do to force Bzlmod to unpack the latest version of Bazel, so it can be enclosed within the image hermetically).

This should be fixed by installing a system-wide `bazel.rc` which includes `--noenable_bzlmod`, running `bazel version`, and then removing the system-wide setting. When this runs within the context of a user's codebase, it is then up to them to either have a `MODULE.bazel` or pass `--noenable_bzlmod` (since this is Bazel's own behavior).

WDYT?

File: images/bazel/tests/02-build.sh [hunk 2]
```diff
@@ -8,4 +8,4 @@ git clone --depth=1 https://github.com/bazelbuild/examples ${TMPDIR}
 docker run --rm \
   -v "${TMPDIR}/flags-parsing-tutorial:/work" \
   -w /work \
-  "${IMAGE_NAME}" build //...
+  "${IMAGE_NAME}" build --noenable_bzlmod //...
```
Comment (sgammon) [inline images/bazel/tests/02-build.sh:11]: This should fix that Bzlmod file issue.

**Comment by sgammon** [comment]
@imjasonh This is ready for review again. Thanks for your patience while I figured that out.

**Comment by sgammon** [comment]
@imjasonh friendly ping :) don't want this to drift, and I'm (patiently) waiting on a merge so I can use this image downstream in our [`build-infra`](https://github.com/elide-dev/build-infra) repo.

**Review by k4leung4 — APPROVED** [review]

File: images/bazel/tests/02-build.sh (hunk 2 above)
Comment (k4leung4) [reply images/bazel/tests/02-build.sh:11]: ack

**Pull request merged**